grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationAtom.java

```diff
@@ -283,6 +293,18 @@ public int computePriority(Set<Var> subbedVars) {
                 .filter(pred -> rolePlayers.contains(pred.getVarName()));
     }
 
+    public Stream<IdPredicate> getRolePredicates(){
+        return getRelationPlayers().stream()
+                .map(RelationPlayer::getRole)
+                .flatMap(CommonUtil::optionalToStream)
+                .filter(var -> var.var().isUserDefinedName())
+                .filter(vp -> vp.getTypeLabel().isPresent())
+                .map(vp -> {
+                    Label label = vp.getTypeLabel().orElse(null);
+                    return new IdPredicate(vp.var(), tx().getRole(label.getValue()), getParentQuery());
```

Comment:
According to the above line: `label.getValue()` is risking a null pointer exception. Might be better to throw a more meaningful error or revisit the logic and see if the null can be avoided.

Reply:
There is a IsPresent above.  The reason I didn't collapse the optional in the stream is that i need both the label and the vp.